docs/design.md

```diff
@@ -82,59 +86,115 @@ library or framework that they instrument. Some examples include:
+When creating this instance the static constructor of the type performs
+the following actions:
+
+- Adds a handler to the
+[`AssemblyResolve` event](https://docs.microsoft.com/en-us/dotnet/api/system.appdomain.assemblyresolve?view=net-5.0),
+so that it can add any assembly needed by the SDK itself or by any instrumentation.
+- Runs, through reflection, the `Initialization` method from
+  the `OpenTelemetry.AutoInstrumentation.Instrumentation` type
+  from the Managed Profiler assembly.
+  - The `Initialization` code bootstraps the OpenTelemetry .NET SDK,
+    adding configured processors, exporters, and so on,
+    and setting the mechanisms to enable any configured source instrumentations.
+
+#### .NET bootstrapping
+
+.NET applications rely on the [host startup hook](https://github.com/dotnet/runtime/blob/main/docs/design/features/host-startup-hook.md)
+being configured to use the `StartupHook.Initialize()` method from
+the `OpenTelemetry.AutoInstrumentation.StartupHook` assembly.
+The `Initialize` method will load the `OpenTelemetry.AutoInstrumentation.Loader`
+assembly and will create the `OpenTelemetry.AutoInstrumentation.Loader.Startup` instance.
+
+#### .NET Framework bootstrapping
+
+.NET Framework bootstrapping is performed on the [CorProfiler::JITCompilationStarted](https://docs.microsoft.com/en-us/dotnet/framework/unmanaged-api/profiling/icorprofilercallback-jitcompilationstarted-method)
+callback. When that callback happens the CLR Profiler DLL takes
+the following actions:
+
+- If this the first user module in the current AppDomain,
+the profiler injects the IL calling the Loader `Startup` constructor.
+- If the first method observed by JITCompilationStarted is IIS startup code,
+  the profiler invokes
+  `AppDomain.CurrentDomain.SetData("OpenTelemetry_IISPreInitStart", true)`,
+  so that automatic instrumentations can correctly handle IIS startup scenarios.
+
+### Enabling and injecting instrumentations
+
+#### Source instrumentation
 
-The OpenTelemetry .NET SDK and selected source instrumentations are injected
-into the target process through a series of steps started by the CLR Profiler DLL:
+Source instrumentations are injected into the target process by adding
+handlers to the [`AssemblyLoad` event](https://learn.microsoft.com/en-us/dotnet/api/system.appdomain.assemblyload?view=net-7.0)
+and when the targeted assembly is loaded triggering the source instrumentation
+initialization code.
+
+#### Bytecode instrumentation
+
+The bytecode instrumentations, called "call target" in this repo, relies on
+the JIT recompilation capability of the CLR to rewrite the IL for instrumented
+methods. This adds logic at the beginning and end of the instrumented methods
+to invoke instrumentation written in this repo, and wraps the calls with
+try-catch blocks to prevent instrumentation errors from affecting the normal operation
+of the application. This IL rewrite happens in the following steps:
```

Comment:
```suggestion
Bytecode instrumentations, also defined as "call target" in this repo, rely on
the JIT recompilation capability of the CLR to rewrite the IL for instrumented
methods. This adds logic at the beginning and end of the instrumented methods
to invoke instrumentation included in this project, and wraps the calls with
try-catch blocks to prevent instrumentation errors from affecting the normal operation
of the application. This IL code rewrite happens in the following steps:
```

Any reason for using two names? Couldn't we just use "bytecode instrumentation" or something instead of "call target"?

Reply:
This is more for historical reasons: there were different forms of bytecode instrumentation in the past, one was "call site" the other "call target". You are right: it is time to remove "call target" from this type of document.

Reply:
a1eee3830